Example/Tests/CallSpec.swift

```diff
@@ -68,6 +68,70 @@ class CallSpec: QuickSpec {
+                                                                                                "Accept-Charset" : "utf-8"])
+                let call = Call(path: "path", method: .GET, rootNode: nil, parameters: params)
+                let request = call.request(withConfiguration: configuration)
+                expect(request?.allHTTPHeaderFields?.keys).to(contain("Accept-Language"))
+                expect(request?.allHTTPHeaderFields?.values).to(contain("utf-8"))
+            }
+            
+            it("should fail to insert http headers that arent strings into the request") {
+                let params = Parameters(type: Parameters.ParameterType.httpHeader, parameters: ["Accept-Language" : 12345,
+                                                                                                "Accept-Charset" : Data(base64Encoded: "el wrongo")])
+                let call = Call(path: "path", method: .GET, rootNode: nil, parameters: params)
+                let request = call.request(withConfiguration: configuration)
+                expect(request?.allHTTPHeaderFields?.keys).toNot(contain("Accept-Language"))
+                expect(request?.allHTTPHeaderFields?.values).toNot(contain("utf-8"))
+            }
+            
+            it("should insert URL components into the request") {
+                let params = Parameters(type: Parameters.ParameterType.urlComponents, parameters: ["some query item": "aa🗿🤔aej"])
+                let call = Call(path: "path", method: .GET, rootNode: nil, parameters: params)
+                let request = call.request(withConfiguration: configuration)
+                expect(request?.url?.absoluteString).to(contain("some%20query%20item=aa%F0%9F%97%BF%F0%9F%A4%94aej"))
+            }
+            
+            it("should fail to insert URL components that arent strings into the request") {
+                let params = Parameters(type: Parameters.ParameterType.urlComponents, parameters: ["some dumb query item": Data(base64Encoded: "el wrongo")])
+                let call = Call(path: "path", method: .GET, rootNode: nil, parameters: params)
+                let request = call.request(withConfiguration: configuration)
+                expect(request?.url?.absoluteString).toNot(contain("some%20dumb%20query%20item"))
+            }
+            
+            it("should add JSON into the request body into the request") {
+                let params = Parameters(type: Parameters.ParameterType.jsonBody, parameters: ["a string": "i am a string. well met.",
+                                                                                              "a number": 123])
+                let call = Call(path: "path", method: .PUT, rootNode: nil, parameters: params)
+                let data = call.request(withConfiguration: configuration)?.httpBody
+                expect(data).toNot(beNil())
+                
+                do {
```

Comment:
🎈In een test moet je geen do catch doen. Try! is goed want moet toch altijd slagen want is een test.